Assets/Scripts/Source/Game/GameFlowManager.cs

```diff
@@ -38,4 +112,53 @@ public void UnloadScene(string sceneName)
         audioManager.Stop("Trees");
         audioManager.Play("Music 1", new AudioOptions(3,0,0.5f,true,false));
     }
+
+    public void RestartLevel()
+    {
+        if (loading)
+            return;
+
+        StartCoroutine(ReloadLevel());
+    }
+
+    public void NextLevel()
+    {
+        if (loading)
+            return;
+
+        currentLevel++;
+        if (currentLevel >= levels.Count)
+        {
+            loading = true;
+            combatUI.DoFadeOut();
+            Debug.Log("You won!");
+        }
+        else
+        {
+            StartCoroutine(ReloadLevel());
+        }
+    }
+
+    IEnumerator ReloadLevel()
+    {
+        loading = true;
+        combatUI.DoFadeOut();
+
+        yield return new WaitForSeconds(1f);
```

Comment:
Is the 1f waiting for the fade out 

If so, could we have `combatUI` return the wait time? Or pass the rest of this logic to it as an `OnComplete` delegate?